pkg/providers/instancetype/instancetype.go

```diff
@@ -65,6 +68,7 @@ type DefaultProvider struct {
 
 	muInstanceTypesOfferings sync.RWMutex
 	instanceTypesOfferings   map[string]sets.Set[string]
+	allZones                 sets.Set[string]
```

Comment:
nit: This is being updated and gated  with a mutex by `muInstanceTypesInfo` -- You should potentially consider putting this up with the lock as a best practice since it's gating access

Reply:
It's being gated by `muInstanceTypesOfferings`, not `Info`.